src/app/components/passwordToggleButton.component.ts

```diff
@@ -0,0 +1,28 @@
+import { PasswordState } from '../components';
+
+export class PasswordToggleButton{
+    public create(passwordInput: HTMLInputElement) : HTMLButtonElement {
+        const showButton = document.createElement('button');
+        showButton.textContent = 'Show';
+
+        showButton.addEventListener('click', (event) => {
+          event.preventDefault();
+          const newState : PasswordState = this.togglePassword(passwordInput.type);
+          showButton.textContent = newState.title;
+          passwordInput.type = newState.type;
+        });
+        return showButton;
+    }
+
+    private togglePassword(buttonType:string) : PasswordState {
+        const newState : PasswordState = {type: "", title: ""};
```

Comment:
Please, check this approach
`
return buttonType === 'password' ? {title: "", type: ""} : {title: "", type: ""}
`

Reply:
done